x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ingest/InferenceProcessor.java

```diff
@@ -38,4 +136,120 @@ public IngestDocument execute(IngestDocument ingestDocument) {
+                    "Adjust the setting [{}]: [{}] if a greater number is desired.",
+                    RestStatus.CONFLICT,
+                    currentInferenceProcessors,
+                    MAX_INFERENCE_PROCESSORS.getKey(),
+                    maxIngestProcessors);
+            }
+
+            String modelId = ConfigurationUtils.readStringProperty(TYPE, tag, config, MODEL_ID);
+            String targetField = ConfigurationUtils.readStringProperty(TYPE, tag, config, TARGET_FIELD);
+            Map<String, String> fieldMapping = ConfigurationUtils.readOptionalMap(TYPE, tag, config, FIELD_MAPPINGS);
+            InferenceConfig inferenceConfig = inferenceConfigFromMap(ConfigurationUtils.readMap(TYPE, tag, config, INFERENCE_CONFIG));
+            String modelInfoField = ConfigurationUtils.readStringProperty(TYPE, tag, config, MODEL_INFO_FIELD, "_model_info");
+            return new InferenceProcessor(client, tag, targetField, modelId, inferenceConfig, fieldMapping, modelInfoField);
+        }
+
+        // Package private for testing
+        void setMaxIngestProcessors(int maxIngestProcessors) {
+            logger.debug("updating setting maxIngestProcessors from [{}] to [{}]", this.maxIngestProcessors, maxIngestProcessors);
+            this.maxIngestProcessors = maxIngestProcessors;
+        }
+
+        InferenceConfig inferenceConfigFromMap(Map<String, Object> inferenceConfig) throws IOException {
+            ExceptionsHelper.requireNonNull(inferenceConfig, INFERENCE_CONFIG);
+
+            if (inferenceConfig.keySet().size() != 1) {
+                throw ExceptionsHelper.badRequestException("{} must be an object with one inference type mapped to an object.",
+                    INFERENCE_CONFIG);
+            }
+            Object value = inferenceConfig.values().iterator().next();
+
+            if ((value instanceof Map<?, ?>) == false) {
+                throw ExceptionsHelper.badRequestException("{} must be an object with one inference type mapped to an object.",
+                    INFERENCE_CONFIG);
+            }
+            @SuppressWarnings("unchecked")
+            Map<String, Object> valueMap = (Map<String, Object>)value;
+
+            if (inferenceConfig.containsKey(ClassificationConfig.NAME)) {
+                checkSupportedVersion(new ClassificationConfig(0));
+                return ClassificationConfig.fromMap(valueMap);
+            } else if (inferenceConfig.containsKey(RegressionConfig.NAME)) {
+                checkSupportedVersion(new RegressionConfig());
+                return RegressionConfig.fromMap(valueMap);
+            } else {
+                throw ExceptionsHelper.badRequestException("unrecognized inference configuration type {}. Supported types {}",
+                    inferenceConfig.keySet(),
+                    Arrays.asList(ClassificationConfig.NAME, RegressionConfig.NAME));
+            }
+        }
+
+        void checkSupportedVersion(InferenceConfig config) {
```

Comment:
Initially, this factory builds the pipeline on the master node before storing it in cluster state. So, if we always check the minimum supported version against the minimum node version we can guarantee that `PUT pipeline` will fail if a user tries to create a pipeline when the specific processor setting is not supported.

Reply:
I think this explanation should go into code comment for method `getMinimalSupportedVersion`.